pkg/client/add_events.go

```diff
@@ -184,5 +185,15 @@ func (client *DataSetClient) SendAllAddEventsBuffers() {
 		return true
 	})
 
-	// TODO: wait on all sending to be finished
+	for client.buffersEnqueued.Load() > client.buffersProcessed.Load() {
+		client.Logger.Info(
+			"Not all buffers has been processed",
+			zap.Uint64("buffersEnqueued", client.buffersEnqueued.Load()),
+			zap.Uint64("buffersProcessed", client.buffersProcessed.Load()),
+		)
+		time.Sleep(client.Config.RetryBase)
```

Comment:
Actually, never mind, that code is inside the for loop and not outside it. So we likely want to use some value > 0 to prevent busy waiting hogging CPU.

In this case, do we need / want some other sleep delay for this specific scenario? Usually it kinds depends on how long we expect this operation to take to complete and values >= 0.1 are used to prevent busy waiting.

Reply:
It's configuration option. We are happy with any value. We are not validating it.

Reply:
Yeah, but it seems like we are (ab)using this retry delay which primary applies to a different use case (retry on api errors and hiccups) vs sleep we use here to avoid busy waiting. Aka we may not want to use as high sleep here as we use in other retry scenarios.